protocols/kad/src/behaviour.rs

```diff
@@ -2702,10 +2787,15 @@ pub type GetProvidersResult = Result<GetProvidersOk, GetProvidersError>;
 
 /// The successful result of [`Kademlia::get_providers`].
 #[derive(Debug, Clone)]
-pub struct GetProvidersOk {
-    pub key: record::Key,
-    pub providers: HashSet<PeerId>,
-    pub closest_peers: Vec<PeerId>,
+pub enum GetProvidersOk {
+    FoundProviders {
+        key: record::Key,
+        /// The new set of providers discovered.
+        providers: HashSet<PeerId>,
+    },
+    FinishedWithNoAdditionalRecord {
+        closest_peers: Vec<PeerId>,
```

Comment:
As an aside, while we have a `put_record_to` to put a record to an individual node, we don't have a `put_provider_record_to`. 

That seems surprising to me, but I guess not needed by anyone out there thus far.

(Not to be fixed in this pull request.)